### ISSUE TITLE
tflint failing while enabling API's in a new GCP project

### ISSUE THREAD

**Issue by boreddysandhya** [terraform-linters/tflint#2594]
### Summary

Tflint  check is failing with below error for PR- https://lloydsbanking.ghe.com/internal-developer-platform/a0625a-entdp-orch01-iac-gcp/pull/201

<img width="1914" height="849" alt="Image" src="https://github.com/user-attachments/assets/07e9be5b-de64-444a-abb9-e08ce75dbfeb" />

Run tflint --init
Installing "azurerm" plugin...
Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(25)
 -> 1: runtime.gopanic: /panic.go(860)
 -> 2: runtime.panicmem: /panic.go(336)
 -> 3: runtime.sigpanic: /signal_unix.go(931)
 -> 4: github.com/sigstore/sigstore-go/pkg/bundle.(*Bundle).TlogEntries: /bundle.go(300)
 -> 5: github.com/sigstore/sigstore-go/pkg/verify.VerifyTlogEntry: /tlog.go(40)
 -> 6: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).VerifyTransparencyLogInclusion: /signed_entity.go(809)
 -> 7: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).Verify: /signed_entity.go(606)
 -> 8: github.com/terraform-linters/tflint/plugin.(*SignatureChecker).VerifyAttestations: /signature.go(136)
 -> 9: github.com/terraform-linters/tflint/plugin.(*InstallConfig).Install: /install.go(134)
 -> 10: github.com/terraform-linters/tflint/cmd.(*CLI).init.func1: /init.go(49)
 -> 11: github.com/terraform-linters/tflint/cmd.(*CLI).withinChangedDir: /cli.go(200)
 -> 12: github.com/terraform-linters/tflint/cmd.(*CLI).init: /init.go(23)
 -> 13: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(108)
 -> 14: main.main: /main.go(38)
 -> 15: runtime.main: /proc.go(290)
 -> 16: runtime.goexit: /asm_amd64.s(1771)

TFLint crashed... :(
Please attach an output log, describe the situation and version that occurred and post an issue to https://github.com/terraform-linters/tflint/issues
Error: Process completed with exit code 1.

### Command

tflint

### Terraform Configuration

```terraform
I have added api.tf file in a0625a-entdp-orch01-iac-gcp, Branch EDPDPE-5302. When i raise Pull Request during terraform Plan it ran the PR checks and failing with TFLINT error
```

### TFLint Configuration

```HCL
Not sure from where i can get it
```

### Output

```Shell
Run tflint --init
Installing "azurerm" plugin...
Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(25)
 -> 1: runtime.gopanic: /panic.go(860)
 -> 2: runtime.panicmem: /panic.go(336)
 -> 3: runtime.sigpanic: /signal_unix.go(931)
 -> 4: github.com/sigstore/sigstore-go/pkg/bundle.(*Bundle).TlogEntries: /bundle.go(300)
 -> 5: github.com/sigstore/sigstore-go/pkg/verify.VerifyTlogEntry: /tlog.go(40)
 -> 6: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).VerifyTransparencyLogInclusion: /signed_entity.go(809)
 -> 7: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).Verify: /signed_entity.go(606)
 -> 8: github.com/terraform-linters/tflint/plugin.(*SignatureChecker).VerifyAttestations: /signature.go(136)
 -> 9: github.com/terraform-linters/tflint/plugin.(*InstallConfig).Install: /install.go(134)
 -> 10: github.com/terraform-linters/tflint/cmd.(*CLI).init.func1: /init.go(49)
 -> 11: github.com/terraform-linters/tflint/cmd.(*CLI).withinChangedDir: /cli.go(200)
 -> 12: github.com/terraform-linters/tflint/cmd.(*CLI).init: /init.go(23)
 -> 13: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(108)
 -> 14: main.main: /main.go(38)
 -> 15: runtime.main: /proc.go(290)
 -> 16: runtime.goexit: /asm_amd64.s(1771)

TFLint crashed... :(
Please attach an output log, describe the situation and version that occurred and post an issue to https://github.com/terraform-linters/tflint/issues
Error: Process completed with exit code 1.
```

### TFLint Version

0.63.1

### Operating System

- [ ] Linux
- [ ] macOS
- [ ] Windows

**Comment by DanyloSafankov** [comment]
We have the similar issue on AWS on version `0.61.0`

```
Installing "terraform" plugin...
Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(25)
 -> 1: runtime.gopanic: /panic.go(783)
 -> 2: runtime.panicmem: /panic.go(262)
 -> 3: runtime.sigpanic: /signal_unix.go(925)
 -> 4: github.com/sigstore/sigstore-go/pkg/bundle.(*Bundle).TlogEntries: /bundle.go(300)
 -> 5: github.com/sigstore/sigstore-go/pkg/verify.VerifyTlogEntry: /tlog.go(40)
 -> 6: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).VerifyTransparencyLogInclusion: /signed_entity.go(809)
 -> 7: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).Verify: /signed_entity.go(606)
 -> 8: github.com/terraform-linters/tflint/plugin.(*SignatureChecker).VerifyAttestations: /signature.go(136)
 -> 9: github.com/terraform-linters/tflint/plugin.(*InstallConfig).Install: /install.go(130)
 -> 10: github.com/terraform-linters/tflint/cmd.(*CLI).init.func1: /init.go(49)
 -> 11: github.com/terraform-linters/tflint/cmd.(*CLI).withinChangedDir: /cli.go(200)
 -> 12: github.com/terraform-linters/tflint/cmd.(*CLI).init: /init.go(23)
 -> 13: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(108)
 -> 14: main.main: /main.go(38)
 -> 15: runtime.main: /proc.go(285)
 -> 16: runtime.goexit: /asm_amd64.s(1693)

TFLint crashed... :(
Please attach an output log, describe the situation and version that occurred and post an issue to https://github.com/terraform-linters/tflint/issues
```

**Comment by sramekmichal** [comment]
Hello, same problem here, but for AWS:
```
$ tflint -v
TFLint version 0.63.1
$ tflint --init
Installing "aws" plugin...
Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(25)
 -> 1: runtime.gopanic: /panic.go(860)
 -> 2: runtime.panicmem: /panic.go(336)
 -> 3: runtime.sigpanic: /signal_unix.go(931)
 -> 4: github.com/sigstore/sigstore-go/pkg/bundle.(*Bundle).TlogEntries: /bundle.go(300)
 -> 5: github.com/sigstore/sigstore-go/pkg/verify.VerifyTlogEntry: /tlog.go(40)
 -> 6: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).VerifyTransparencyLogInclusion: /signed_entity.go(809)
 -> 7: github.com/sigstore/sigstore-go/pkg/verify.(*Verifier).Verify: /signed_entity.go(606)
 -> 8: github.com/terraform-linters/tflint/plugin.(*SignatureChecker).VerifyAttestations: /signature.go(136)
 -> 9: github.com/terraform-linters/tflint/plugin.(*InstallConfig).Install: /install.go(134)
 -> 10: github.com/terraform-linters/tflint/cmd.(*CLI).init.func1: /init.go(49)
 -> 11: github.com/terraform-linters/tflint/cmd.(*CLI).withinChangedDir: /cli.go(200)
 -> 12: github.com/terraform-linters/tflint/cmd.(*CLI).init: /init.go(23)
 -> 13: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(108)
 -> 14: main.main: /main.go(38)
 -> 15: runtime.main: /proc.go(290)
 -> 16: runtime.goexit: /asm_amd64.s(1771)
TFLint crashed... :(
Please attach an output log, describe the situation and version that occurred and post an issue to https://github.com/terraform-linters/tflint/issues
```

**Comment by kilianpaquier** [comment]
Hello, a working fix is to add `signature = "pgp"` to plugins definition within `tflint.hcl`.

See working comment and associated issue: https://github.com/terraform-linters/tflint/issues/2591#issuecomment-4996273694

**Comment by nsballmann** [comment]
> Hello, a working fix is to add `signature = "pgp"` to plugins definition within `tflint.hcl`.
> 
> See working comment and associated issue: [#2591 (comment)](https://github.com/terraform-linters/tflint/issues/2591#issuecomment-4996273694)

But this also creates output like this with the most recent versions:

```log
Installing "terraform" plugin...
The plugin was signed using a legacy PGP signing key. Please update the plugin to the latest version
Installed "terraform" (source: github.com/terraform-linters/tflint-ruleset-terraform, version: 0.15.0)
Installing "aws" plugin...
The plugin was signed using a legacy PGP signing key. Please update the plugin to the latest version
Installed "aws" (source: github.com/terraform-linters/tflint-ruleset-aws, version: 0.48.0)
Installing "azurerm" plugin...
The plugin was signed using a legacy PGP signing key. Please update the plugin to the latest version
Installed "azurerm" (source: github.com/terraform-linters/tflint-ruleset-azurerm, version: 0.32.0)
```

This indicates to me that while it may help it seems to be "only" a workaround that may relay on a legacy solution.

I assume that the latest plugins are currently signed by a "legacy key". Maybe the previous signing key lifetime expired and that's the reason it is considered a legacy signing key (now)?

**Comment by baer95** [comment]
@nsballmann the linked issue explains what is happening. tl;dr: the default signature (GitHub Release Attestation) was changed by GitHub API, which is causing this breakage.

Its probably good to keep the discussion in one place, lets continue in the linked issue :)

**Comment by nsballmann** [comment]
> [@nsballmann](https://github.com/nsballmann) the linked issue explains what is happening. tl;dr: the default signature (GitHub Release Attestation) was changed by GitHub API, which is causing this breakage.
> 
> Its probably good to keep the discussion in one place, lets continue in the linked issue :)

Ah true, I came in from #2596 and overlooked that the link is to a third issue, my mistake.
